cadence/transactions/gieft/claim_nft.cdc

```diff
@@ -0,0 +1,46 @@
+import "Giefts"
+import "NonFungibleToken"
+import "ExampleNFT"
+import "MetadataViews"
+
+// This transaction attempts to claim an nft from a gieft
+// and deposit it into the authorizer's public collection
+
+// @param gieftOwner: The address of the gieft owner
+// @param gieftID: The ID of the gieft to claim an nft from
+// @param password: The password of the gieft to claim an nft from
+
+transaction(gieftOwner: Address, gieftID: UInt64, password: String) {
+
+    let gieftPublic: &Giefts.GieftCollection{Giefts.GieftCollectionPublic}
+    let collectionPublic: Capability<&ExampleNFT.Collection{ExampleNFT.ExampleNFTCollectionPublic}>
+
+    prepare(acct: AuthAccount) {
+        // Initialize the NFT collection if it doesn't exist
```

Comment:
For this transaction to scale to all types of NFTs in the catalog I think we should withdraw the nft first, then try to resolve its collectionData view and init the collection based on that. We use this pattern in .find quite a lot and it works really well.